### DISCUSSION TITLE
CSS style is not correctly being applied when I do conditional render

### DISCUSSION THREAD

**Discussion by TianboZ** [vercel/next.js#17155] [Help] [answered]
I need to conditionally render components based on screen size. Since I am using getInitialProps for data fetching,  the page is server-side rendered. As for as I know, the only way to detect device type on server side is via user-agent string, but it is not most reliable way since devices are changing fast. So I plan to detect device screen size on client side. So I implement a customized hook to do it. 

useWindowSize.js

```javascript
import { useEffect, useState } from 'react';

export default function useWindowSize() {
    const [windowSize, setWindowSize] = useState({
        width: typeof window === 'undefined' ? 1200 : window.innerWidth, // default width 1200
    });

    useEffect(() => {
        // Handler to call on window resize
        function handleResize() {
            // Set window width/height to state
            setWindowSize({
                width: window.innerWidth,
                //height: window.innerHeight,
            });
        }

        // Add event listener
        window.addEventListener('resize', handleResize);

        // Call handler right away so state gets updated with initial window size
        handleResize();

        // Remove event listener on cleanup
        return () => window.removeEventListener('resize', handleResize);
    }, []); // Empty array ensures that effect is only run on mount

    return windowSize.width <= 600;
}
```

then I use this hook to detect window size and conditional render components

```javascript
export default function IndexPage() {
  const isMobile = useWindowSize();

  if (typeof window !== "undefined") {
    // if you are running it on codesanbox, I don't know why log is not printed
    console.log("client side re-render");
  }

  return (
    <div>
      {isMobile ? (
        <div
          style={{
            color: "red",
            fontSize: 40
          }}
        >
          mobile
        </div>
      ) : (
        <div
          style={{
            color: "blue",
            fontSize: 20
          }}
        >
          desktop
        </div>
      )}
    </div>
  );
}

IndexPage.getInitialProps = () => {
  return {
    a: 1
  };
};

```

when I load the page on mobile browser, you will see 

![image](https://user-images.githubusercontent.com/28853847/93391997-e86a1980-f824-11ea-9ad0-a5424c73986f.png)

text `mobile` is applied wrong CSS style. video demo: https://share.getcloudapp.com/nOuk08L0

**how to reproduce:**
https://codesandbox.io/s/thirsty-khayyam-npqpt

Can someone please help me out. Thank you in advance!

**Comment by TianboZ** [comment, marked as answer]
Does anyone have any idea?

**Reply by pederklev** [reply]
The codesandbox seems to work as intended for me. I get large, red text when I resize the window to narrower than 600px, and smaller blue on desktop.

Anyway, things I would do to troubleshoot:
- Check what happens when setting the default width returned in useWindowSize to something mobile-sized. For instance, 500.
- Check what happens if you put your CSS inside styled-jsx tags, and apply classnames to your components instead. Check if there's any difference between putting these tags within the conditionally rendered part of the page or not. ```{condition ? <div className="red">red <style jsx>{`.red{color:red}`}</style></div> : <div className="blue">blue <style jsx>{`.blue{color:blue}`}</style></div>}``` vs ```<>{condition ? <div className="red">red </div> : <div className="blue">blue</div>}<style jsx>{`.red{color:red} .blue{color:blue}`}</style></>```
- Look specifically at what is returned within the document sent to the browser, and what style etc is applied before rehydration. You can disable javascript to check this, or look at the document within the Network tab in devtools in Chrome.
- Are there differences between `npm run dev` and `npm run build`?

**Reply by 20manas** [reply]
Hey @TianboZ , how did you resolve this?
